### ISSUE TITLE
Creating battle_test_utility.h/.c file for functions that can be used in all battle and NPC test files

### ISSUE THREAD

**Issue by nmkhan100** [uchicago-cs/chiventure#1533]
There are a number of functions included in all of the battle and NPC test files that are used in all of the tests. Rather than having to copy and paste these functions into each battle test file in order to use them, it would be helpful for there to be a single file with all of these functions that we can simply import into our test files. It would also help to rename a few functions, remove duplicates, and organize and label them in groups in order to make the file more readable (as well as add comments describing functions).

For this issue, I will be creating a battle_test_utility.h/.c file to define functions that create test classes, battle items, stats, etc., so that all of these functions can be used across all of the battle and NPC test files when the file is imported.

**Comment by nmkhan100** [comment]
I'm currently looking through all the battle test files for functions that I can put in my battle_test_utility.c file.

**Comment by nmkhan100** [comment]
I've added all of the functions from the battle test files to my battle_test_utility.c file.

**Comment by nmkhan100** [comment]
I'm looking through all the NPC test files for functions that I can put in my battle_test_utility.c file.

**Comment by nmkhan100** [comment]
I've added all of the functions from the NPC test files to my battle_test_utility.c file.

**Comment by nmkhan100** [comment]
I've removed all of the duplicate functions, renamed a few of the functions since they were too unclear, and both organized and labeled all the functions into groups. I've also added modified/new comments describing each function because they were either badly described or nonexistent.

**Comment by nmkhan100** [comment]
I made sure the header file was included in every battle and NPC test file and that I modified the functions used in the test files (in case I removed duplicate functions that have different names or simply renamed certain functions so they have better names).

**Comment by nmkhan100** [comment]
I have created a battle_test_utility.c/.h file for both the battle and NPC tests to import and use functions from. This has been implemented to prevent teams from having to copy paste functions into every test file for the tests to use the functions and pass. The files have also been made more readable through the renaming of different functions, removal of duplicate functions found in multiple test files as a result of copy pasting, and the grouping of similar functions (and labeling). I have also both added and modified comments describing all the functions. Now it will be very easy for these teams to use functions in their tests, and in the future, if they want to add more functions for the tests to use, they can simply add it to these files I have created.
